src/iuse.cpp

```diff
@@ -8068,6 +8068,36 @@ int iuse::hairkit(player *p, item *it, bool, const tripoint&)
     return it->type->charges_to_use();
 }
 
+int iuse::washclothes( player *p, item *it, bool, const tripoint& )
+{
+    if( it->charges < it->type->charges_to_use() ) {
+        p->add_msg_if_player( _( "You need soap to use this." ) );
+    } else if ( g->m.has_nearby_water( g->u.pos() ) ) {
+        int pos = g->inv_for_filter( _( "Wash what?" ), []( const item & itm ) {
+            return itm.has_flag( "FILTHY" ); } );
+    item *mod = &( p->i_at( pos ) );
+
+    if( mod == nullptr ) {
+        p->add_msg_if_player( m_info, _( "You don't have that item!" ) );
+        return 0;
+    }
+
+    if ( !mod->has_flag( "FILTHY" ) ) {
```

Comment:
Oh, yes. I just started work on this PR prior to your inventory refactoring and forgot to remove this check.